### PR TITLE
[BACKLOG-6228] Adding update calculated measure annotation

### PR DESCRIPTION
**Pull request by pminutillo**

### CODE REVIEW

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Dpentaho.coding.standards.version=1.0.1 -Djunit.jvmargs=-XX:PermSize=256m -f build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  src/org/pentaho/agilebi/modeler/messages/messages_en_US.properties
  src/org/pentaho/agilebi/modeler/models/annotations/ModelAnnotation.java
  src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMember.java
  src/org/pentaho/agilebi/modeler/models/annotations/util/AnnotationConstants.java
  src/org/pentaho/agilebi/modeler/models/annotations/util/MondrianSchemaHandler.java
  src/org/pentaho/agilebi/modeler/models/annotations/util/XMLUtil.java
  test-res/products.with.calc.measures.mondrian.xml
  test-src/org/pentaho/agilebi/modeler/models/annotations/AnnotationUtil.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/CreateCalculatedMemberIT.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/ModelAnnotationGroupIT.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMemberTest.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/util/MondrianSchemaHandlerTest.java


Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards

File src/org/pentaho/agilebi/modeler/models/annotations/util/MondrianSchemaHandler.java
1. Message: Unused import - org.apache.xpath.operations.Bool. at Line: 27 Column: 8
2. Message: 'if' is not followed by whitespace. at Line: 435 Column: 11
3. Message: '}' at column 9 should be on the same line as the next part of a multi-block statement. at Line: 440 Column: 9

File src/org/pentaho/agilebi/modeler/models/annotations/util/XMLUtil.java
1. Message: '}' at column 5 should be on the same line as the next part of a multi-block statement. at Line: 104 Column: 5
2. Message: '(' is preceded with whitespace. at Line: 99 Column: 43

File test-src/org/pentaho/agilebi/modeler/models/annotations/util/MondrianSchemaHandlerTest.java
1. Message: 'if' is not followed by whitespace. at Line: 383 Column: 9
2. Message: '{' is not preceded with whitespace. at Line: 387 Column: 72





Newly Fixed Tests:
-----------------
org.pentaho.agilebi.modeler.models.annotations.CreateCalculatedMemberIT.testApplyMondrian: junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError: 
	at org.pentaho.agilebi.modeler.models.annotations.CreateCalculatedMemberIT.testApplyMondrian(CreateCalculatedMemberIT.java:149)

```



Unit test coverage change:
============
org.pentaho.agilebi.modeler.models.annotations.util.MondrianSchemaHandler:
 * instructionPercentCoverageChange : +2.63%
 * branchPercentCoverageChange :**-7.14%**
 * linePercentCoverageChange : +3.01%
 * complexityPercentCoverageChange :**-10.78%**
 * methodPercentCoverageChange : +1.67%
    
org.pentaho.agilebi.modeler.models.annotations.UpdateCalculatedMember:
 * branchPercentCoverage :60.00%
 * linePercentCoverage :84.00%
 * methodPercentCoverage :78.95%
 * complexityPercentCoverage :66.67%
 * instructionPercentCoverage :79.64%
    
org.pentaho.agilebi.modeler.models.annotations.ModelAnnotation.Type:
 * instructionPercentCoverageChange : +0.84%
 * linePercentCoverageChange : +1.08%
    
org.pentaho.agilebi.modeler.models.annotations.util.XMLUtil:
 * branchPercentCoverage :100.00%
 * instructionPercentCoverageChange : +9.26%
 * linePercentCoverageChange : +9.52%
 * complexityPercentCoverageChange : +11.43%
 * methodPercentCoverageChange : +6.67%
    
Integration test coverage change:
============
org.pentaho.agilebi.modeler.models.annotations.util.MondrianSchemaHandler:
 * instructionPercentCoverageChange :**-9.55%**
 * branchPercentCoverageChange :**-11.69%**
 * linePercentCoverageChange :**-9.21%**
 * complexityPercentCoverageChange :**-5.27%**
 * methodPercentCoverageChange :**-2.08%**
    
org.pentaho.agilebi.modeler.models.annotations.ModelAnnotation.Type:
 * instructionPercentCoverageChange : +1.28%
 * linePercentCoverageChange : +1.30%

**Comment by lucboudreau** [comment]
A couple classes are loosing ~10% coverage with this PR. Please check if this can be addressed.

**Comment by pminutillo** [comment]
@lucboudreau Taking a look

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Dpentaho.coding.standards.version=1.0.1 -Djunit.jvmargs=-XX:PermSize=256m -f build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  src/org/pentaho/agilebi/modeler/messages/messages_en_US.properties
  src/org/pentaho/agilebi/modeler/models/annotations/ModelAnnotation.java
  src/org/pentaho/agilebi/modeler/models/annotations/RemoveMeasure.java
  src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMember.java
  src/org/pentaho/agilebi/modeler/models/annotations/util/AnnotationConstants.java
  src/org/pentaho/agilebi/modeler/models/annotations/util/MondrianSchemaHandler.java
  src/org/pentaho/agilebi/modeler/models/annotations/util/XMLUtil.java
  test-res/products.with.calc.measures.mondrian.xml
  test-src/org/pentaho/agilebi/modeler/models/annotations/AnnotationUtil.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/CreateCalculatedMemberIT.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/ModelAnnotationGroupIT.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMemberTest.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/util/MondrianSchemaHandlerTest.java






Newly Fixed Tests:
-----------------
org.pentaho.agilebi.modeler.models.annotations.CreateCalculatedMemberIT.testApplyMondrian: junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError: 
	at org.pentaho.agilebi.modeler.models.annotations.CreateCalculatedMemberIT.testApplyMondrian(CreateCalculatedMemberIT.java:149)

```



Unit test coverage change:
============
org.pentaho.agilebi.modeler.ModelerException:
 * instructionPercentCoverageChange : +25.00%
 * linePercentCoverageChange : +25.00%
 * complexityPercentCoverageChange : +25.00%
 * methodPercentCoverageChange : +25.00%
    
org.pentaho.agilebi.modeler.models.annotations.util.MondrianSchemaHandler:
 * instructionPercentCoverageChange : +15.19%
 * branchPercentCoverageChange : +7.63%
 * linePercentCoverageChange : +18.70%
 * complexityPercentCoverageChange : +17.30%
 * methodPercentCoverageChange : +26.67%
    
org.pentaho.agilebi.modeler.models.annotations.UpdateCalculatedMember:
 * branchPercentCoverage :60.00%
 * linePercentCoverage :84.00%
 * methodPercentCoverage :78.95%
 * complexityPercentCoverage :66.67%
 * instructionPercentCoverage :79.64%
    
org.pentaho.agilebi.modeler.models.annotations.ModelAnnotation.Type:
 * instructionPercentCoverageChange : +0.84%
 * linePercentCoverageChange : +1.08%
    
org.pentaho.agilebi.modeler.models.annotations.util.XMLUtil:
 * branchPercentCoverage :100.00%
 * instructionPercentCoverageChange : +9.26%
 * linePercentCoverageChange : +9.52%
 * complexityPercentCoverageChange : +11.43%
 * methodPercentCoverageChange : +6.67%
    
Integration test coverage change:
============
org.pentaho.agilebi.modeler.models.annotations.util.MondrianSchemaHandler:
 * instructionPercentCoverageChange :**-9.22%**
 * branchPercentCoverageChange :**-11.69%**
 * linePercentCoverageChange :**-8.35%**
 * complexityPercentCoverageChange :**-5.05%**
    
org.pentaho.agilebi.modeler.models.annotations.ModelAnnotation.Type:
 * instructionPercentCoverageChange : +1.28%
 * linePercentCoverageChange : +1.30%

**Comment by pminutillo** [comment]
@kurtwalker @rfellows @DFieldFL @kolinus @bennychow I am still working on updating the integration test coverage, but please feel free to review if you have a chance

**Comment by kurtwalker** [comment]
:+1:

**Comment by bennychow** [comment]
The UpdateCalculatedMember.java class is missing the currencySymbol set by the user in the UI.  This is also needed to re-hydrate the edit calculated members dialog.

**Comment by pminutillo** [comment]
@bennychow We do not have any fields in the existing create calculated measure dialog for the user to set the currency symbol, and the CreateCalculatedMember annotation does not have the currencySymbol either. I assumed it was derived from a combination of the selected format category and possibly a localization setting. I am taking a closer look now.

**Comment by rfellows** [comment]
@bennychow The calculated measure dialog does not have a currency symbol input field. I assume this relates to the bug you filed (http://jira.pentaho.com/browse/BACKLOG-5886). I spent a quick minute looking into that one. it seems that currency symbols were overlooked when building out the create calc dialog (for reports and inline annotations). I commented on the bug... basically stating that this should be considered a new feature and sprinted on in the future.

**Comment by pminutillo** [comment]
@rfellows ah, thanks Rob!

**Comment by rfellows** [comment]
@pminutillo is this ready to go? are your tests updated? i can't finish the front-end without the class. if it's going to be a while, i can just grab your fork and build locally.

**Comment by pminutillo** [comment]
@rfellows Yeah my tests have been updated. I was waiting on the platform-core dependency issue to be resolved so I could run another wingman analysis, but all test coverage should be up to par

File: src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMember.java [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+import org.w3c.dom.Document;
+
+import java.util.logging.Logger;
+
+/**
+ * Annotation to create a calculated member
+ * <p/>
+ * Created by pminutillo on 2/2/15.
+ */
+@MetaStoreElementType( name = "CreatedCalculatedMember", description = "CreatedCalculatedMember Annotation" )
```
Comment (rfellows) [inline src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMember.java]: shouldn't the name & description reflect that this is an update, not create?

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Dpentaho.coding.standards.version=1.0.1 -Djunit.jvmargs=-XX:PermSize=256m -f build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  src/org/pentaho/agilebi/modeler/messages/messages_en_US.properties
  src/org/pentaho/agilebi/modeler/models/annotations/ModelAnnotation.java
  src/org/pentaho/agilebi/modeler/models/annotations/RemoveMeasure.java
  src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMember.java
  src/org/pentaho/agilebi/modeler/models/annotations/util/AnnotationConstants.java
  src/org/pentaho/agilebi/modeler/models/annotations/util/MondrianSchemaHandler.java
  src/org/pentaho/agilebi/modeler/models/annotations/util/XMLUtil.java
  test-res/products.with.calc.measures.mondrian.xml
  test-src/org/pentaho/agilebi/modeler/models/annotations/AnnotationUtil.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/CreateCalculatedMemberIT.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/ModelAnnotationGroupIT.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMemberTest.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/util/MondrianSchemaHandlerTest.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/util/MondrianSchemaHandlerTestIT.java






Newly Fixed Tests:
-----------------
org.pentaho.agilebi.modeler.models.annotations.CreateCalculatedMemberIT.testApplyMondrian: junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError: 
	at org.pentaho.agilebi.modeler.models.annotations.CreateCalculatedMemberIT.testApplyMondrian(CreateCalculatedMemberIT.java:149)

```



Unit test coverage change:
============
org.pentaho.agilebi.modeler.ModelerException:
 * instructionPercentCoverageChange : +25.00%
 * linePercentCoverageChange : +25.00%
 * complexityPercentCoverageChange : +25.00%
 * methodPercentCoverageChange : +25.00%
    
org.pentaho.agilebi.modeler.models.annotations.util.MondrianSchemaHandler:
 * instructionPercentCoverageChange : +15.19%
 * branchPercentCoverageChange : +7.63%
 * linePercentCoverageChange : +18.70%
 * complexityPercentCoverageChange : +17.30%
 * methodPercentCoverageChange : +26.67%
    
org.pentaho.agilebi.modeler.models.annotations.UpdateCalculatedMember:
 * branchPercentCoverage :60.00%
 * linePercentCoverage :84.00%
 * methodPercentCoverage :78.95%
 * complexityPercentCoverage :66.67%
 * instructionPercentCoverage :79.64%
    
org.pentaho.agilebi.modeler.models.annotations.ModelAnnotation.Type:
 * instructionPercentCoverageChange : +0.84%
 * linePercentCoverageChange : +1.08%
    
org.pentaho.agilebi.modeler.models.annotations.util.XMLUtil:
 * branchPercentCoverage :100.00%
 * instructionPercentCoverageChange : +9.26%
 * linePercentCoverageChange : +9.52%
 * complexityPercentCoverageChange : +11.43%
 * methodPercentCoverageChange : +6.67%
    
Integration test coverage change:
============
org.pentaho.agilebi.modeler.models.annotations.util.MondrianSchemaHandler:
 * instructionPercentCoverageChange : +35.84%
 * branchPercentCoverageChange : +33.77%
 * linePercentCoverageChange : +38.52%
 * complexityPercentCoverageChange : +39.02%
 * methodPercentCoverageChange : +40.00%
    
org.pentaho.agilebi.modeler.models.annotations.ModelAnnotation.Type:
 * instructionPercentCoverageChange : +1.28%
 * linePercentCoverageChange : +1.30%
    
org.pentaho.agilebi.modeler.models.annotations.util.XMLUtil:
 * branchPercentCoverage :100.00%
 * linePercentCoverage :28.57%
 * methodPercentCoverage :16.67%
 * complexityPercentCoverage :28.57%
 * instructionPercentCoverage :25.00%

File: src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMember.java [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+
+    MondrianSchemaHandler mondrianSchemaHandler = new MondrianSchemaHandler( schema );
+
+    MondrianDef.CalculatedMember calculatedMember = new MondrianDef.CalculatedMember();
+    calculatedMember.name = this.getName();
+    calculatedMember.caption = this.getCaption();
+    calculatedMember.description = this.getDescription();
+    calculatedMember.dimension = this.getDimension();
+    calculatedMember.formula = this.getFormula();
+    calculatedMember.visible = this.isHidden();
```
Comment (rfellows) [inline src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMember.java]: should be !this.isHidden()

File: src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMember.java [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+    calculatedMember.memberProperties = new MondrianDef.CalculatedMemberProperty[] { solveOrder };
+
+    // annotation to indicate this was created via inline modeling, should always be true
+    MondrianDef.Annotation inline = new MondrianDef.Annotation();
+    inline.name = "AnalyzerInlineCreatedInline";
+    inline.cdata = isInline() ? "true" : "false";
+
+    // annotation to store decimal places (scale)
+    MondrianDef.Annotation formatScaleAnnotation = new MondrianDef.Annotation();
+    formatScaleAnnotation.name = "formatScale";
```
Comment (rfellows) [inline src/org/pentaho/agilebi/modeler/models/annotations/UpdateCalculatedMember.java]: shouldn't these be the same as in create calc measure?
https://github.com/rfellows/modeler/blob/c1498e7c9e686f032308eed8ee0a4fb46ec3e3d5/src/org/pentaho/agilebi/modeler/models/annotations/CreateCalculatedMember.java#L224

```
AnnotationConstants.INLINE_ANNOTATION_FORMAT_SCALE
```

**Pull request merged**